internal/helm/controller/controller.go

```diff
@@ -81,7 +82,7 @@ func Add(mgr manager.Manager, options WatchOptions) error {
 	o := &unstructured.Unstructured{}
 	o.SetGroupVersionKind(options.GVK)
 
-	if err := c.Watch(source.Kind(mgr.GetCache(), o), &libhandler.InstrumentedEnqueueRequestForObject{}); err != nil {
+	if err := c.Watch(source.Kind(mgr.GetCache(), client.Object(o), &libhandler.InstrumentedEnqueueRequestForObject[client.Object]{})); err != nil {
```

Comment:
This looks wrong to me. Feels like it should be...

```suggestion
	if err := c.Watch(source.Kind(mgr.GetCache(), o, &libhandler.InstrumentedEnqueueRequestForObject[*unstructured.Unstructured]{})); err != nil {
```

Reply:
The way the code is written here is the same as in ansible, it does look strange, but it seems to function the same way. Is there a functionality issue or just an ascetics one? I'd rather be consistent on this across repos.

Reply:
Going back to the original code for consistency.